### PR TITLE
Inline directory removal check into cleanup

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- inline directory existence check into `_cleanup_temporary_directory`
- remove obsolete `_should_remove_created_dir` helper
- adjust cleanup unit test for early-return behavior

## Testing
- `make fmt`
- `make lint`
- `make typecheck`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_68a4fc1878d08322ad6ce7154a0d0626

## Summary by Sourcery

Inline the directory removal predicate into the cleanup method, remove the now-obsolete helper, and update tests for the new early-return behavior

Enhancements:
- Inline directory existence check directly into `_cleanup_temporary_directory` and remove the `_should_remove_created_dir` helper
- Add an early-exit path in cleanup to skip removal when no matching directory exists

Documentation:
- Expand the cleanup method docstring to detail its idempotent and conditional removal behavior

Tests:
- Remove tests for the deleted helper and adjust the cleanup test to reflect the new early-return logic

## Summary by CodeRabbit

- **Bug Fixes**
  - More reliable temporary-directory cleanup: skips removal when the directory is missing or replaced, safely deletes only when conditions are met, and always clears internal tracking.

- **Refactor**
  - Consolidated cleanup logic for clearer, idempotent behavior.

- **Tests**
  - Expanded coverage for cleanup edge cases and failure handling with new helpers and parameterized tests.

- **Documentation**
  - Minor README formatting cleanup for Windows note.

### CODE REVIEW

File: cmd_mox/unittests/test_environment.py [hunk 1]
```diff
@@ -295,44 +295,12 @@ def test_environment_manager_cleanup_error_handling(
...
-def test_should_remove_created_dir_returns_false_when_never_entered() -> None:
-    """Return False if EnvironmentManager was never entered."""
-    env = EnvironmentManager()
-    assert not env._should_remove_created_dir()
-
-
-def test_cleanup_temporary_directory_skips_when_predicate_false() -> None:
-    """Skip directory removal if predicate returns False."""
+def test_cleanup_temporary_directory_skips_when_no_directory() -> None:
+    """Skip directory removal when no directory was created or it's gone."""
```
[the remark this answers is not in the document]
Comment (leynos) [reply cmd_mox/unittests/test_environment.py:299]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Add a unit test for the scenario where `shim_dir` is changed or replaced after creation to ensure the early-exit logic in `_cleanup_temporary_directory` still works as intended.
- Extract the combined boolean check into a small private helper or named variable to improve readability and make each condition more self-documenting.
- Consider condensing the `_cleanup_temporary_directory` docstring to the core behaviors to reduce verbosity and keep maintenance easier.
